File: 70. Climbing Stairs/climbing stairs.cpp
using namespace std;
//#include <math.h>
/*
* Warning : This solution is too slow, it cannot pass the OJ of Leetcode
*/


class Solution 
{
public:
	int climbStairs(int n) 
	{
		if (n == 1)
			return 1;
		if (n == 2)
			return 2;
		return climbStairs(n - 1) + climbStairs(n - 2);
	}
};

File: 70. Climbing Stairs/climbing stairs.py
__author__ = 'Bill Quan'
import math

class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        count = 0
        for j in range(0, n/2 + 1):
            i = n - 2 * j
            count += math.factorial(i+j) / (math.factorial(i) * math.factorial(j))
        return count

print Solution.climbStairs(Solution(), 4)


